core/src/main/java/org/elasticsearch/action/admin/indices/analyze/AnalyzeRequest.java

```diff
@@ -136,6 +171,9 @@ public void readFrom(StreamInput in) throws IOException {
         tokenFilters = in.readStringArray();
         charFilters = in.readStringArray();
         field = in.readOptionalString();
+        detail = in.readBoolean();
```

Comment:
Did you miss Ryan's comment. You need to check the version before writing or reading the value.